crypto/evp/evp_fetch.c

```diff
@@ -51,6 +53,67 @@ struct method_data_st {
     void (*destruct_method)(void *method);
 };
 
+static int add_names_to_namemap(OSSL_NAMEMAP *namemap,
+                                const char *names)
+{
+    const char *p, *q;
+    size_t l;
+    int id = 0;
+
+    /* Check that we have a namemap and that there is at least one name */
+    if (namemap == NULL) {
+        ERR_raise(ERR_LIB_EVP, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+
+    /*
+     * Check that no name is an empty string, and that all names have at
+     * most one numeric identity together.
+     */
+    for (p = names; *p != '\0'; p = q == NULL ? p + l : q + 1) {
```

Comment:
I recommend some parenthesis in the `p = q == NULL ....` to make it clearer.

Reply:
Sure. Only around the conditional, or around the whole ?: expression?

Reply:
I'd do it: `p = (q == NULL ? p + l : q + 1)`

Since I got it in the wrong reply.

Reply:
Makes sense. Thanks

Reply:
Done